bigtable/google/cloud/bigtable/table.py

```diff
@@ -363,6 +375,69 @@ def sample_row_keys(self):
         return response_iterator
 
 
+class _MutateRowsRetryableError(Exception):
+    """A retryable error in Mutate Rows response."""
+
+    def __init__(self, retryable_responses=None):
+        super(_MutateRowsRetryableError, self).__init__()
+        self.retryable_responses = retryable_responses or []
+
+
+class _RetryableMutateRowsWorker(object):
+    # pylint: disable=unsubscriptable-object
+    RETRY_CODES = (
+        StatusCode.DEADLINE_EXCEEDED.value[0],
+        StatusCode.ABORTED.value[0],
+        StatusCode.UNAVAILABLE.value[0],
+    )
+
+    def __init__(self, client, table_name, rows):
+        self.client = client
+        self.table_name = table_name
+        self.rows = rows
+        self.responses_statuses = [
+            None for _ in six.moves.xrange(len(self.rows))]
+
+    def __call__(self):
+        return self._do_mutate_retryable_rows()
+
+    def _is_retryable(self, status):  # pylint: disable=no-self-use
+        return (status is None or
+                status.code in _RetryableMutateRowsWorker.RETRY_CODES)
+
+    def _do_mutate_retryable_rows(self):
+        retryable_rows = []
+        index_into_all_rows = []
+        for i, status in enumerate(self.responses_statuses):
+            if self._is_retryable(status):
+                retryable_rows.append(self.rows[i])
+                index_into_all_rows.append(i)
+
+        if not retryable_rows:
+            # All mutations are either successful or non-retryable now.
+            return self.responses_statuses
+
+        mutate_rows_request = _mutate_rows_request(
+            self.table_name, retryable_rows)
+        responses = self.client._data_stub.MutateRows(
+            mutate_rows_request)
+
+        retryable_responses = []
+        for response in responses:
+            for entry in response.entries:
+                index = index_into_all_rows[entry.index]
+                self.responses_statuses[index] = entry.status
+                if self._is_retryable(entry.status):
+                    entry.index = index  # Override with index into all rows
+                    retryable_responses.append(entry)
+                if entry.status.code == 0:
+                    self.rows[index].clear()
+
+        if retryable_responses:
+            raise _MutateRowsRetryableError(retryable_responses)
```

Comment:
I think this will lose errors that are not retryable

Reply:
This exception is raised when there are retryable errors, so the caller can call again if they want to attempt retry. 
Otherwise, the callable returns a complete `responses_statuses` that contains the success, retryable and non-retryable statuses for corresponding rows.

Reply:
Oh, I see, I didn't realize that this error was only used for flow control and wasn't bubbled up to the user.